target/linux/ramips/dts/mt7621.dtsi

```diff
@@ -505,43 +558,59 @@
 			0x01000000 0 0x00000000 0x1e160000 0 0x00010000 /* io space */
 		>;
 
-		interrupt-parent = <&gic>;
-		interrupts = <GIC_SHARED 4 IRQ_TYPE_LEVEL_HIGH
-				GIC_SHARED 24 IRQ_TYPE_LEVEL_HIGH
-				GIC_SHARED 25 IRQ_TYPE_LEVEL_HIGH>;
+		#interrupt-cells = <1>;
+		interrupt-map-mask = <0xF0000 0 0 1>;
+		interrupt-map = <0x10000 0 0 1 &gic GIC_SHARED 4 IRQ_TYPE_LEVEL_HIGH>,
+				<0x20000 0 0 1 &gic GIC_SHARED 24 IRQ_TYPE_LEVEL_HIGH>,
+				<0x30000 0 0 1 &gic GIC_SHARED 25 IRQ_TYPE_LEVEL_HIGH>;
 
 		status = "disabled";
 
 		resets = <&rstctrl 24 &rstctrl 25 &rstctrl 26>;
 		reset-names = "pcie0", "pcie1", "pcie2";
 		clocks = <&clkctrl 24 &clkctrl 25 &clkctrl 26>;
 		clock-names = "pcie0", "pcie1", "pcie2";
+		phys = <&pcie0_phy 1>, <&pcie2_phy 0>;
+		phy-names = "pcie-phy0", "pcie-phy2";
+
+		reset-gpios = <&gpio 19 GPIO_ACTIVE_LOW>,
+				<&gpio 8 GPIO_ACTIVE_LOW>,
+				<&gpio 7 GPIO_ACTIVE_LOW>;
```

Comment:
Only gpio19 should be included here because this is the only pin used as pcie reset by all boards. gpio8 and gpio7 should be added to device dts separately if it's actually needed for a specific device.

Reply:
The device tree you sent me the other day was using this three gpios in this file in previous 4.14 version and that was the reason of this change. Why is this different now?

Reply:
As mentioned in the application note, there are two possible way of wiring pcie reset:
1. connect gpio19 to all pcie reset pins
2. connect gpio19 to pcie0 reset and pick two other gpios for pcie1 and pcie2

gpio7 and gpio8 may not be used as pcie reset. And consider that we never used gpio7 and gpio8 as reset in 4.14 anyway, we can safely assume that most vendor choosed 1. instead of 2.

> The device tree you sent me the other day was using this three gpios in this file

Vendor can do all kinds of hacky stuff in their SDK and this doesn't mean we should follow them. A gpio pin can't be used as pcie reset and led/buttons simutaneously and we should only request gpio pins actually used by pcie instead of occupying unused pins.

Reply:
Ok, I have sent this for the change pointed out here:

http://driverdev.linuxdriverproject.org/pipermail/driverdev-devel/2020-March/142567.html